includes/admin/class-wp-job-manager-writepanels.php

```diff
@@ -585,14 +585,23 @@ public function save_job_listing_data( $post_id, $post ) {
 		/* Set Post Status To Expired If Already Expired */
 		$expiry_date = get_post_meta( $post_id, '_job_expires', true );
 		$today_date  = date( 'Y-m-d', current_time( 'timestamp' ) );
-		$post_status = $expiry_date && $today_date > $expiry_date ? 'expired' : false;
-		if( $post_status ) {
+		$expire_date_in_past = $expiry_date && $today_date > $expiry_date;
+		if( $expire_date_in_past ) {
 			remove_action( 'job_manager_save_job_listing', array( $this, 'save_job_listing_data' ), 20, 2 );
-			$job_data = array(
-				'ID'          => $post_id,
-				'post_status' => $post_status,
-			);
-			wp_update_post( $job_data );
+			if (
+				isset( $_POST['post_status'] )
```

Comment:
This can be made more intention-revealing if we extract method (something like `is_job_listing_reactivated` or similar)